swarm/src/lib.rs

```diff
@@ -524,14 +521,6 @@ where
                         return Err(DialError::DialPeerConditionFalse(condition));
                     }
 
-                    // Check if peer is banned.
-                    if self.banned_peers.contains(&peer_id) {
-                        let error = DialError::Banned;
-                        #[allow(deprecated)]
-                        self.behaviour.inject_dial_failure(Some(peer_id), &error);
-                        return Err(error);
-                    }
```

Comment:
We lose this early check here, i.e. we will initiate a dial and then immediately drop the connection instead of aborting the connection early. I guess that is the price for implementing this kind of functionality in a generic way.